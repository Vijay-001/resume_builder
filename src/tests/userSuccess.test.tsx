import UserCheckout from "../containers/userSuccess/successPage";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import renderer from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react";

const mockStore = configureMockStore();
const store = mockStore({});

describe("<BuildUserResume/>", () => {
  const Component = () => (
    <Provider store={store}>
      <UserCheckout />
    </Provider>
  );

  it("renders correctly", () => {
    const tree = renderer.create(<Component />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders default state values", () => {
    const { getByTestId } = render(<Component />);
    const downloadButton = getByTestId("downloadButton");
    fireEvent.click(downloadButton);
  });
});
