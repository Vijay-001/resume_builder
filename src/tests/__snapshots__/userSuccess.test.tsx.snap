// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BuildUserResume/> renders correctly 1`] = `
Array [
  <h5
    className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
  >
    Thank you for build resume.
  </h5>,
  <button
    className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    data-testid="downloadButton"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Download
  </button>,
]
`;
